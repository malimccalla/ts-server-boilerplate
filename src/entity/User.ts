import * as bcrypt from 'bcryptjs';
import {
  BaseEntity,
  BeforeInsert,
  Column,
  Entity,
  PrimaryGeneratedColumn,
} from 'typeorm';

@Entity('users')
export class User extends BaseEntity {
  @PrimaryGeneratedColumn('uuid') id: string;

  @Column('varchar', { length: 255 })
  email: string;

  @Column('text') password: string;

  @Column('boolean', { default: false })
  confirmed: boolean;

  @Column('boolean', { default: false })
  locked: boolean;

  @BeforeInsert()
  async hashPassword() {
    const password = await bcrypt.hash(this.password, 10);
    this.password = password;
  }
}
