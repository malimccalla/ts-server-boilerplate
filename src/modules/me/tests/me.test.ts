import { Connection } from 'typeorm';

import { User } from '../../../entity/User';
import { createTestConn } from '../../../test/createTestConn';
import { TestClient } from '../../../test/TestClient';

const host = process.env.TEST_HOST as string;

describe('me', () => {
  const email = 'cookie@email.com';
  const password = 'password123';
  let conn: Connection;

  beforeEach(async () => {
    conn = await createTestConn();

    await User.create({
      email,
      password,
      confirmed: true,
    }).save();
  });

  afterEach(async () => {
    await conn.close();
  });

  test('should return null if no cookie', async () => {
    const client = new TestClient(host);
    const response = await client.me();

    expect(response.data.me).toBeNull();
  });

  test('get the current user', async () => {
    const client = new TestClient(host);
    await client.login(email, password);

    const res = await client.me();

    expect(res.data).toMatchSnapshot();
    expect(res.data.me.email).toEqual(email);
  });
});
