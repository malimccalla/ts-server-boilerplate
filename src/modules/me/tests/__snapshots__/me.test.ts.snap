// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`me get the current user 1`] = `
Object {
  "me": Object {
    "email": "cookie@email.com",
    "id": "d33602aa-893d-4cbc-8061-c359b84891f5",
  },
}
`;
